PeriodogramAnalysis/PowerSpectrumAnalysis.py

```diff
@@ -59,139 +56,147 @@ def __call__(self, signal: SignalType):
+
+        Parameters:
+        -----------
+        output : tuple
+            Output from the __call__ method, containing spectrogram data dictionary.
+        show : bool, optional (default=False)
+            If set to True, the spectrogram will be displayed.
+        save_path : str, optional (default=None)
+            If provided, the spectrogram plot will be saved to the given path with filenames indicating the chunk and channel.
+        """
+        num_of_chunks = self.chunk_idx + 1
+        spec_dict = output[1]
+
         for chunk in range(num_of_chunks):
             for channel in range(self.num_channels):
-                if channel not in exclude_channel_list:
-                    spectrogram_data = spec_dict[chunk][channel]
-                    frequencies = spectrogram_data['frequencies']
-                    times = spectrogram_data['times']
-                    Sxx = spectrogram_data['Sxx']
-
-                    plt.figure(figsize=(10, 6))
-                    plt.pcolormesh(times, frequencies, 10 * np.log10(Sxx), shading='gouraud')
-                    plt.colorbar(label='Power spectral density (dB/Hz)')
-                    plt.xlabel('Time (s)')
-                    plt.ylabel('Frequency (Hz)')
-                    plt.title(f'Spectrogram for Channel {channel}')
-                    if frequency_limit:
-                        plt.ylim(frequency_limit)
-                    if show:
-                        plt.show()
-                    if save_path is not None:
-                        fig_save_path = os.path.join(save_path, "Analysis_figures")
-                        os.makedirs(fig_save_path, exist_ok=True)
-                        plot_path = os.path.join(fig_save_path, f'Spectrogram_Channel_{channel}.png')
-                        plt.savefig(plot_path, dpi=300)
-                    plt.close()
+                if channel in self.exclude_channel_list:
+                    continue
+                spectrogram_data = spec_dict[chunk][channel]
+                frequencies = spectrogram_data['frequencies']
+                times = spectrogram_data['times']
+                Sxx = spectrogram_data['Sxx']
+
+                plt.figure(figsize=(10, 6))
+                plt.pcolormesh(times, frequencies, 10 * np.log10(Sxx), shading='gouraud')
+                plt.colorbar(label='Power spectral density (dB/Hz)')
+                plt.xlabel('Time (s)')
+                plt.ylabel('Frequency (Hz)')
+                plt.title(f'Spectrogram for Channel {channel}')
+                plt.ylim(self.frequency_limit)
+
+                if show:
+                    plt.show()
+                if save_path is not None:
+                    plot_path = os.path.join(save_path, f'Chunk{chunk}:Spectrogram_Channel_{channel}.png')
+                    plt.savefig(plot_path, dpi=300)
+                plt.close()
 
 def multitaper_psd(x, sfreq, fmin=0.0, fmax=np.inf, bandwidth=None, adaptive=True, low_bias=True):
```

Comment:
This function can be on separate file.

Reply:
I have removed it to another file named MultitaperPowerSpectrum.py